includes/class-wc-payments.php

```diff
@@ -1544,7 +1544,7 @@ public static function remove_woo_admin_notes() {
 	 * @return bool Normal WCPay behavior (false, default) or TRUE if the site should only use network-wide saved payment methods.
 	 */
 	public static function is_network_saved_cards_enabled() {
-		return apply_filters( 'wcpay_force_network_saved_cards', false );
+		return Payment_Method::CARD === self::$card_gateway->get_selected_stripe_payment_type_id() && apply_filters( 'wcpay_force_network_saved_cards', false );
```

Comment:
I agree this makes sense, but unfortunately, `get_selected_stripe_payment_type_id()` from the `$card_gateway` will always return `card`. I agree that semantically this method seems to be able to return the currently selected payment method, but it doesn't. 

My proposal would be to keep `is_network_saved_cards_enabled()` as it is, without changes, and move the check against card payment method to the place where we assign value to `$settings[ $payment_method_id ]['forceNetworkSavedCards']` in `class-wc-payments-checkout.php`